[blog] Sunsetting Intel macOS instances (#1099)
cirruslabs/cirrus-ci-docs: docs/blog/posts/2022-11-08-sunsetting-intel-macos-instances.md

---
draft: false
date: 2022-11-08
links:
  - blog/posts/2018-06-26-announcing-macos-support-on-cirrus-ci.md
  - blog/posts/2020-12-18-persistent-workers.md
  - blog/posts/2021-01-26-new-macos-task-execution-architecture.md
  - blog/posts/2022-07-07-isolating-network-between-tarts-macos-virtual-machines.md
authors:
  - fkorotkov
categories:
  - announcement
  - macos
---

# Sunsetting Intel macOS instances

**TLDR** Intel-based Big Sur and High Sierra instances will stop working on January 1st 2023. Please migrate to [M1-based Monterey and Ventura instances](/guide/macOS.md).
Below we'll provide some history and motivation for this decision.

## Evolution of macOS infrastructure for Cirrus CI

We've been running macOS instances for almost 5 years now. We evaluated all the existing solutions and even successfully
operated two of them on Intel platform before creating our own virtualization toolset for Apple Silicon called [*Tart*](https://github.com/cirruslabs/tart).

> We are switching [managed-by-us macOS instances](/guide/macOS.md)) to exclusively running in Tart virtual machines **starting January 1st 2023**.

Reviewer: ```suggestion
We are switching [managed-by-us macOS instances](/guide/macOS.md) to exclusively running in Tart virtual machines **starting January 1st 2023**.
```

Then in the end of 2019 - early 2020 there were a bunch of transient issues with Anka's networking layer. Sometimes some hosts were just loosing
internet connections and all consecutive Anka VMs were not able to run anything until a restart of a host. We spent countless hours with Veertu folks
trying to debug this transient but very annoying issue with no luck. In the end we had to implement some workaround and detections on our end.
At this point we started thinking of a way to replace Anka Controller, so we could potentially switch the virtualization layer as well.

With that in mind we started working on [Cirrus CLI](https://github.com/cirruslabs/cirrus-cli) -- a CI-agnostic tool that can run "tasks" locally in containers or VMs.

### Second generation with Parallels

![](/blog/images/new-architecture-workers.png)

Throughout 2020, we switched from an Anka cluster managed by MacStadium to a self-managed installation. We deployed
Anka Registry and Anka Controller on Google Cloud and got Mac Minis evenly distributed between two [`MacMiniVault`](https://www.macminivault.com/) data centers for redundancy.

> We perfected our Ansible cookbooks and got very comfortable with rolling updates so we don't have downtime. Prepared

Reviewer: ```suggestion
We perfected our Ansible cookbooks and got very comfortable with rolling updates so we don't have downtime. We also prepared
```

Packer templates to automate creation of Virtual Machines.

In parallel Cirrus CLI matured, it was able to run tasks in Docker containers. It was time to find a replacement for Anka.
We had two criteria in mind: cost-efficiency and network stability. After some research we ended up with Parallels.
Network performance was better, starting time for VMs was a little slower but still very fast. And price! Anka's
license costed us more than we paid for the hardware we rented to run it! Parallels was just $10/month/host.

Long story short, we added necessary features to Cirrus CLI to run tasks in Parallels VMs, used the same Packer templates
to rebuild all the virtual machines. And in early 2021 did the switch!

### Third generation with Tart

<img src="https://github.com/cirruslabs/tart/raw/main/Resources/TartSocial.png"/>

In the meantime Apple Silicon was taking off. It was clear Apple was very serious about the transition and full switch from Intel processors.
But at the time none of the virtualization solutions supported Apple Silicon. It was a new stack with new challenges.

> Thankfully in the end of 2021 with macOS Monterey release Apple themselves released `Virtualization.Framework` so companies like

Reviewer: ```suggestion
Thankfully in the end of 2021 with macOS Monterey release Apple themselves released `Virtualization.Framework`, so companies like
```

Veertu and Parallels don't need to re-invent the wheel and reverse engineer all the things about macOS.

By February 2022 we were getting more and more requests to support M1 workloads in our CI but none of the virtualization

> solution adopted `Virtualization.Framework` beside Anka 3.0. A switch back was off the table. Anka pricing was

Reviewer: ```suggestion
solution adopted `Virtualization.Framework`, except for Anka 3.0. A switch back was off the table. Anka pricing was
```
